lib/trip_dispatcher.rb

```diff
@@ -87,15 +95,77 @@ def load_trips
         trips << trip
       end
 
-      trips
+      return trips
+    end # load_trips
+
+    def next_driver
+      # Array of available drivers
+      all_available = @drivers.find_all do |driver|
+        # Minnie (ID 100) is avail, and no current trips,
+        # but assignment doesnt want us to choose her.
+        driver.trips.length > 0 && driver.status == :AVAILABLE
+      end
+
+      return nil if all_available == []
+
```

Comment:
I like the choice to return `nil` here, instead of raising an exception. This gives whoever called this method the power to raise an exception if needed, but doesn't force it.